packages/contractkit/src/wallets/signers/azure-hsm-signer.ts

```diff
@@ -0,0 +1,63 @@
+// @ts-ignore-next-line
+import { account as Account, bytes as Bytes, hash as Hash, nat as Nat, RLP } from 'eth-lib'
+// @ts-ignore-next-line
+import * as helpers from 'web3-core-helpers'
+// @ts-ignore-next-line
+import { BN } from 'bn.js'
+import { AzureKeyVaultClient } from './azure-key-vault-client'
+import { trimLeading0x } from '@celo/utils/lib/address'
+import * as ethUtil from 'ethereumjs-util'
+import { EIP712TypedData, generateTypedDataHash } from '../../utils/sign-typed-data-utils'
+import { getHashFromEncoded, RLPEncodedTx, toPaddedBuffer } from '../../utils/signing-utils'
+import { Signer } from './signer'
+
+/**
+ * Signs the EVM transaction using an HSM key in Azure Key Vault
+ */
+export class AzureHSMSigner implements Signer {
+  private keyVaultClient: AzureKeyVaultClient
+  private keyName: string
+
+  constructor(keyVaultClient: AzureKeyVaultClient, keyName: string) {
+    this.keyVaultClient = keyVaultClient
+    this.keyName = keyName
+  }
+
+  async signTransaction(
+    addToV: number,
+    encodedTx: RLPEncodedTx
+  ): Promise<{ v: number; r: Buffer; s: Buffer }> {
+    const hash = getHashFromEncoded(encodedTx.rlpEncode)
+    const bufferedMessage = Buffer.from(trimLeading0x(hash), 'hex')
+    const signature = await this.keyVaultClient.signMessage(bufferedMessage, this.keyName)
+
+    const sigV = Nat.fromString(Bytes.fromNumber(addToV + signature.v))
```

Comment:
I've consolidated the logic as much as I could across all three implementations. Most of the common logic is in wallet.ts. There is some small duplicate logic that allows us to have a stronger interface (ex. using Tx type instead of string). The main reason there are separate classes for each signer, is because each underlying library (ledger, azure, and ethUtil) all have different interfaces.